src/sentry/rules/processor.py

```diff
@@ -150,9 +149,12 @@ def condition_matches(
             self.logger.warning("Unregistered condition %r", condition["id"])
             return None
 
-        condition_inst: EventCondition = condition_cls(self.project, data=condition, rule=rule)
+        condition_inst = condition_cls(self.project, data=condition, rule=rule)
         passes: bool = safe_execute(
-            condition_inst.passes, self.event, state, _with_transaction=False
+            condition_inst.passes,  # type: ignore # passes changes depending on the condition type, which each have their own class that inherits from EventCondition, which itself inherits from RuleBase
```

Comment:
is there a way to cast `condition_inst` so that it can be one of any of the event condition classes (ie. `FirstSeenEventCondition`, `EveryEventCondition`, etc.)? same for `action_inst` below